File: app/repositories/admin_actions.py
from sqlalchemy.ext.asyncio import AsyncSession

from app.db.models import AdminAction


class AdminActionsRepository:
    def __init__(self, session: AsyncSession) -> None:
        self.session = session

    async def create(
        self,
        *,
        admin_telegram_id: int,
        action: str,
        target_telegram_id: int | None = None,
    ) -> AdminAction:
        item = AdminAction(
            admin_telegram_id=admin_telegram_id,
            action=action,
            target_telegram_id=target_telegram_id,
        )
        self.session.add(item)
        await self.session.flush()
        return item

File: src/repositories/adminActions.ts
import type { DatabaseTx } from "@/db/client";
import { adminActions } from "@/db/schema";

export interface CreateAdminActionInput {
  adminTelegramId: number;
  action: string;
  targetTelegramId?: number | null;
}

export interface AdminActionsRepository {
  create(input: CreateAdminActionInput): Promise<void>;
}

export function createAdminActionsRepository(tx: DatabaseTx): AdminActionsRepository {
  return {
    async create(input): Promise<void> {
      await tx.insert(adminActions).values({
        adminTelegramId: input.adminTelegramId,
        action: input.action,
        targetTelegramId: input.targetTelegramId ?? null,
      });
    },
  };
}
